### DISCUSSION TITLE
Conversion from Hexidecimal?

### DISCUSSION THREAD

**Discussion by lloydtabb** [duckdb/duckdb#3198] [General]
Hey there.

Some of the other Looker founders and I are working on [Malloy](https://github.com/looker-open-source/[malloy](https://github.com/looker-open-source/malloy/)/), a new experimental data analytics language that eventually translates to SQL.  We are looking into DuckDB support (nice job!).

I can't seem to find a way to convert a hexidecimal string into a number.  Most other dialects allow hexidecimal literals and casting.  For example I would expect.

```
SELECT '0xA'::BIGINT as ten
```
To return 10.

One of our core features is something called [Symmetric Aggregates](https://help.looker.com/hc/en-us/articles/360023722974-A-Simple-Explanation-of-Symmetric-Aggregates-or-Why-On-Earth-Does-My-SQL-Look-Like-That-) and the we are currently blocked.

Alternatively, making a version of MD5 that returns a HUGEINT would also work for us.

Any ideas?

thanks,

**Comment by hannes** [comment]
Both are things we should support, but as far as I know they are not implemented yet. The second issue particularly should be rather simple though.

**Comment by hannes** [comment]
Maybe @samansmink can hack the int-returning md5 together quickly?

**Reply by samansmink** [reply]
sure! https://github.com/duckdb/duckdb/pull/3208

**Comment by Mytherin** [comment]
Hey lloyd, great that you are enjoying DuckDB!

> I can't seem to find a way to convert a hexidecimal string into a number. Most other dialects allow hexidecimal literals and casting. For example I would expect.

> SELECT '0xA'::BIGINT as ten

> To return 10.

I have tried this in Postgres, MySQL and SQLite and none of them seem to like this (SQLite and MySQL both truncate the number and return 0, Postgres throws an error).

Implementing hexadecimal conversion seems like a good idea to me, but as we strive for good compatibility with other database systems I am curious how you are doing this in the other systems/which functions they have to support this conversion that we are missing?

Thanks!

**Reply by lloydtabb** [reply]
Sorry to disappear on you all.  Somehow I'm not getting notification on this discussion.

The int128 or decimal returning MD5 is is a perfect solution.  

Thanks, excited to get this working...
